### ISSUE TITLE
ES Modules

### ISSUE THREAD

**Issue by lastmjs** [jsverify/jsverify#240]
I just want to bring up that with ES Modules becoming standard in all browsers and NodeJS, it would be nice for JSVerify to consider adopting them even now. With the require statements everywhere, I'm having difficulty using JSVerify with my front-end and back-end code. I don't use RequireJS or Browserify, I'm using tools that assume ES Modules support, which is very future proof compared to any kind of require statements.

I suppose the biggest thing I'd like to bring up is could there be a compiled version of the library distributed that will work in all environments? This is a very computational-focused library, I don't see why it can't be compiled down to plain ES 5 or ES 6. The standalone file in the dist folder seemed like it had that purpose, but it has require statements, which I would not consider portable.

Just my thoughts as I try to incorporate JSVerify more into my projects. Thanks!

**Comment by lastmjs** [comment]
How would you feel about a PR for this?

**Comment by phadej** [comment]
Go ahead, I won't do it myself.

I'm not familiar enough with ES modules. There are some dynamic tricky-hackery in `jsverify` to make modules with less boilerplate. If that can be translated to ES modules, than why not.

Note: seems that es modules are supported starting with Node 8.5, and current LTS release is 8.9.1, so there isn't problem on that side. However, you might need to update `.travis.yml`.

**Comment by phadej** [comment]
Sorry, undo:

> --experimental-modules 

I don't like that, for now even node-4.4 or so works without extra flags (see `.travis.yml`). I'd like to preserve it. ES modules time isn't just yet, I'm sorry.

**Comment by lastmjs** [comment]
I agree, the time is not right. I didn't realize it was still behind a flag.

**Comment by brucou** [comment]
what is the status on that a year later?
